test/zmq.qtest

```diff
@@ -435,6 +440,19 @@ class ZmqTest inherits QUnit::Test {
         # timeout test
         reader.setRecvTimeout(1ms);
         assertThrows("ZSOCKET-TIMEOUT-ERROR", \reader.recvMsg());
+
+        # issue #2944: test ZSocketSub.subscribe()
+        ZSocketSub reader2(zctx);
+        reader2.connect(writer.endpoint());
+        reader2.subscribe(Subscription);
+
+        writer.send(hwm);
+        msg = reader.recvMsg();
+        assertEq(2, msg.size());
+        assertEq(Subscription.size() + HelloWorld.size(), msg.contentSize());
+        msg = reader2.recvMsg();
+        assertEq(2, msg.size());
+        assertEq(Subscription.size() + HelloWorld.size(), msg.contentSize());
```

Comment:
this test case tests just the subscribe part and it only tests the success case, isn't it? I would add such a test where `subs_utf8.set(subs, QCS_UTF8, xsink)` will return a 0 (which is an error) and thus NOTHING would be returned from `.subscribe` but `subscribeIntern` wouldn't be called, what do you think?

Reply:
I added a test for this, but it just tests an `ENCODING-CONVERSION-ERROR` and never calls `subscribeIntern()` as you said, because the input is invalid